src/common/collection.c

```diff
@@ -1211,19 +1211,67 @@ gboolean dt_collection_hint_message_internal(void *message)
 
 void dt_collection_hint_message(const dt_collection_t *collection)
 {
+  /* if relevant, determine offset of selection */
+  GList *selected_imgids = dt_collection_get_selected(collection, 1);
+  int selected_imgids_count = g_list_length(selected_imgids);
+  int selected = -1;
+
+  if(selected_imgids_count > 0)
+  {
+    selected = GPOINTER_TO_INT(g_list_nth_data(selected_imgids, 0));
+    selected = dt_collection_image_offset_with_collection(collection, selected);
+    selected++;
+  }
   /* collection hinting */
   gchar *message;
 
   int c = dt_collection_get_count(collection);
   int cs = dt_collection_get_selected_count(collection);
 
-  message = g_strdup_printf(ngettext("%d image of %d in current collection is selected",
-                                     "%d images of %d in current collection are selected", cs),
-                            cs, c);
+  if(cs == 1)
+  {
+    message = g_strdup_printf("%d image of %d (#%d) in current collection is selected", cs, c, selected);
+  }
+  else
+  {
+    message = g_strdup_printf("%d images of %d in current collection are selected", cs, c);
+  }
 
   g_idle_add(dt_collection_hint_message_internal, message);
 }
 
+int dt_collection_image_offset_with_collection(const dt_collection_t *collection, int imgid)
+{
+  const gchar *qin = dt_collection_get_query(collection);
+  int offset = 0;
+  sqlite3_stmt *stmt;
+
+  if(qin)
+  {
+    DT_DEBUG_SQLITE3_PREPARE_V2(dt_database_get(darktable.db), qin, -1, &stmt, NULL);
+    DT_DEBUG_SQLITE3_BIND_INT(stmt, 1, 0);
+    DT_DEBUG_SQLITE3_BIND_INT(stmt, 2, -1);
+
+    gboolean found = FALSE;
+
+    while(sqlite3_step(stmt) == SQLITE_ROW)
+    {
+      int id = sqlite3_column_int(stmt, 0);
+      if(imgid == id)
+      {
+        found = TRUE;
+        break;
+      }
+      offset++;
+    }
+
+    if(!found) offset = 0;
+
+    sqlite3_finalize(stmt);
+  }
+  return offset;
+}
+
 int dt_collection_image_offset(int imgid)
```

Comment:
Make `dt_collection_image_offset()` just be `return dt_collection_image_offset_with_collection(darktable.collection, imgid);` so that the implementation of the two won't diverge.